feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.ts

```diff
@@ -0,0 +1,208 @@
+  multipleSelectionValues: SelectionValue[] = [];
+
+  ngOnInit() {
+    if (this.attribute.values && this.attribute.values.length > 0) {
+      this.multipleSelectionValues = this.attribute.values.map(
+        ({ name, quantity, selected, valueCode }) => ({
+          name,
+          quantity,
+          selected,
+          valueCode,
+        })
+      );
+    }
+
+    if (
+      this.attribute?.required &&
+      this.multipleSelectionValues.filter((value) => value.selected).length < 2
+    ) {
+      this.preventAction$.next(true);
+    }
+  }
+
+  /**
+   * Updates the value dependent on the provided state
+   *
+   * @param  {any} valueCode - value code to be updated
+   * @param  {any} state - selected state
+   *
+   * @return {ConfigFormUpdateEvent} - form update event
+   */
+  protected updateMultipleSelectionValues(
+    valueCode: any,
+    state: any
+  ): ConfigFormUpdateEvent {
+    const index = this.multipleSelectionValues.findIndex(
+      (value) => value.valueCode === valueCode
+    );
+
+    this.multipleSelectionValues[index] = {
+      ...this.multipleSelectionValues[index],
+      selected: state,
+    };
+
+    const event: ConfigFormUpdateEvent = {
+      changedAttribute: {
+        ...this.attribute,
+        values: this.multipleSelectionValues,
+      },
+      ownerKey: this.ownerKey,
+      updateType: Configurator.UpdateType.ATTRIBUTE,
+    };
+
+    return event;
+  }
+
+  /**
+   * Updates the quantity of the given value
+   *
+   * @param  eventValue - event value
+   *
+   * @return {ConfigFormUpdateEvent} - form update event
+   */
+  protected updateMultipleSelectionValuesQuantity(eventValue: {
+    valueCode: string;
+    quantity: number;
+  }): ConfigFormUpdateEvent | undefined {
+    const value: Configurator.Value = this.multipleSelectionValues.find(
+      (selectionValue) => selectionValue?.valueCode === eventValue.valueCode
+    );
+
+    if (!value) return;
+
+    value.quantity = eventValue.quantity;
+
+    const event: ConfigFormUpdateEvent = {
+      changedAttribute: {
+        ...this.attribute,
+        values: [value],
+      },
+      ownerKey: this.ownerKey,
+      updateType: Configurator.UpdateType.VALUE_QUANTITY,
+    };
+
+    return event;
+  }
+
+  onHandleAttributeQuantity(quantity: any): void {
+    this.loading$.next(true);
+
+    const event: ConfigFormUpdateEvent = {
+      changedAttribute: {
+        ...this.attribute,
+        quantity,
+      },
+      ownerKey: this.ownerKey,
+      updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
+    };
+
+    this.selectionChange.emit(event);
+  }
+
+  onSelect(eventValue: any): void {
+    this.loading$.next(true);
+    this.selectionChange.emit(
+      this.updateMultipleSelectionValues(eventValue, true)
+    );
+  }
+
+  onDeselect(eventValue: any): void {
+    this.loading$.next(true);
+    this.selectionChange.emit(
+      this.updateMultipleSelectionValues(eventValue, false)
+    );
+  }
+
+  onDeselectAll(): void {
```

Comment:
we can also remove this function, just call `onSelect` and but not pass `valueCode`

Reply:
@AlexanderKuschner, could you have a look at it?

Reply:
As discussed: addressed with #12057